server/src/main/java/keywhiz/service/resources/automation/v2/BatchResource.java

```diff
@@ -120,27 +130,38 @@ public BatchCreateOrUpdateSecretsResponseV2 batchCreateOrUpdateSecrets(
   private void createOrUpdateSecret(
       DSLContext dslContext,
       AutomationClient automationClient,
-      CreateOrUpdateSecretInfoV2 secret) {
+      CreateOrUpdateSecretInfoV2 secret,
+      SecretController.SecretBuilder builder) {
 
-    Optional<SecretSeriesAndContent> maybeSecretSeriesAndContent = secretDAO.getSecretByName(dslContext, secret.name());
     String secretOwner = secret.owner();
-    if (maybeSecretSeriesAndContent.isPresent()) {
-      permissionCheck.checkAllowedOrThrow(automationClient, Action.UPDATE, maybeSecretSeriesAndContent.get());
-    } else {
-      permissionCheck.checkAllowedForTargetTypeOrThrow(automationClient, Action.CREATE, Secret.class);
-      secretOwner = getSecretOwnerForSecretCreation(dslContext, secretOwner, automationClient);
+    {
```

Comment:
It was originally to prove to myself (and the IDE) that the contents of that block were isolated, since I was toying with moving it to a separate method in order to reduce clutter. In the end I kept the scope but decided against the method.